src/middlewares/authMiddleware.js

```diff
@@ -0,0 +1,21 @@
+const jwtService = require('../services/jwt.service');
+const { ApiError } = require('../utils/api.error');
+
+function authMiddleware(req, res, next) {
+  const authHeader = req.headers['authorization'] || '';
+  const [, accessToken] = authHeader.split(' ');
+
+  if (!authHeader || !accessToken) {
+    throw ApiError.unauthorized();
+  }
+
+  const userData = jwtService.validateAccessToken(accessToken);
+
+  if (!userData) {
+    throw ApiError.unauthorized();
+  }
+
+  next();
```

Comment:
Consider attaching the `userData` to the `req` object (e.g., `req.user = userData;`) after successful validation. This allows downstream middleware and route handlers to access the authenticated user's information.